Engine/TestPlanExecution.cs

```diff
@@ -95,10 +95,16 @@ enum failState
 
         internal static void PrintWaitingMessage(IEnumerable<IResource> resources)
         {
+            // Save disconnected ressources to avoid race conditions.
+            var waitingRessources = resources.Where(r => !r.IsConnected).ToList();
```

Comment:
Let's use ToArray here - if the list is empty then an empty array gets us Array.Empty<Resource>(), which does not cause any allocation.

Reply:
Now uses .ToArray()